### ISSUE TITLE
openmaptiles/import-osm:0.5 unable to parse planet-181105.osm.pbf

### ISSUE THREAD

**Issue by EX0l0N** [openmaptiles/openmaptiles#528]
## Context
We are using the docker image `openmaptiles/import-osm:0.5` to import a full `planet.osm.pbf`.

While this succeeded with older `planet.osm.pbf` files in august and semptember (2018), we were unable to use newer files.

`imposm3` always stops in the first phase while quick scanning the whole file.

```
Importing in normal mode
[Nov  8 20:20:11] [INFO] [reader] reading /import/planet.osm.pbf with data till 2018-11-05 02:00:01 +0000 UTC
[Nov  8 20:21:11] [INFO] [  1m0s] C: 4040000/s (241040000) N:   38200/s (2283520) W:       0/s (0) R:      0/s (0)
[Nov  8 20:22:11] [INFO] [  2m0s] C: 3852000/s (460928000) N:   54600/s (6542271) W:       0/s (0) R:      0/s (0)
panic: runtime error: index out of range3968000/s (649424000) N:   56300/s  332100/s (9733215) W:       0/s       0/s (0) R:      0/s      0/s (0)

goroutine 52 [running]:
github.com/omniscale/imposm3/parser/pbf.readDenseNodes(0xc456307f80, 0xc444c10000, 0xc45bfca000, 0x4048, 0x4048, 0xc42b108000, 0x3e80, 0x3e80, 0xc43176a000, 0x2af, ...)
        /go/src/github.com/omniscale/imposm3/parser/pbf/pbf.go:35 +0x51e
github.com/omniscale/imposm3/parser/pbf.(*Parser).parseBlock(0xc42026c3c0, 0x7ffebba12d5b, 0x16, 0x17bb8cf0e, 0x472432)
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:96 +0x594
github.com/omniscale/imposm3/parser/pbf.(*Parser).Parse.func1(0xc420223980, 0xc42026c3c0)
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:56 +0xc8
created by github.com/omniscale/imposm3/parser/pbf.(*Parser).Parse
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:65 +0x101
```

## Expected Behavior
While there may be some corrupted data inside the official `planet.osm.pbf`, `imposm3` should skip that and not die because of out of bounds array access.

## Actual Behavior
`imposm3` dies at parser/pbf.readDenseNodes:35

## Possible Fix
- Check slice length before addressing arbitrary fields.
- Use newer version of `imposm3`
- I'm currently unaware why you forked `imposm3` and if this can be reproduced with the current version of the original. I will try use the current version as replacement in the docker container, but I'm not sure if it's compatible.

## Steps to Reproduce
1. Download: `https://planet.osm.org/pbf/planet-181105.osm.pbf`
2. Clone `https://github.com/openmaptiles/openmaptiles`
3. Put `planet-181105.osm.pbf` as `planet.osm.pbf` in `openmaptiles` `/data` dir
4. Run `./quickstart planet`

**Comment by EX0l0N** [comment]
This also happens with an alternate extract with stripped metadata:

http://download.bbbike.org/osm/planet/planet-latest-nometa.osm.pbf

The stacktrace looks slightly different though:
```
Importing in normal mode
[Nov  9 14:25:29] [INFO] removing existing cache /cache
[Nov  9 14:25:29] [INFO] [reader] reading /import/planet.osm.pbf with data till 2018-11-05 02:00:01 +0000 UTC
[Nov  9 14:26:29] [INFO] [  1m0s] C: 4244000/s (227293611) N:   41300/s (2356691) W:       0/s (0) R:      0/s (0)
[Nov  9 14:27:29] [INFO] [  2m0s] C: 4023000/s (454846225) N:   57200/s (6699725) W:       0/s (0) R:      0/s (0)
[Nov  9 14:28:29] [INFO] [  3m0s] C: 4043000/s (701715086) N:   59000/s (10484653) W:       0/s (0) R:      0/s (0)
[Nov  9 14:29:29] [INFO] [  4m0s] C: 4154000/s (970181264) N:   61100/s (14517213) W:       0/s (0) R:      0/s (0)
[Nov  9 14:30:29] [INFO] [  5m0s] C: 4228000/s (1237004171) N:   57100/s (16970523) W:       0/s (0) R:      0/s (0)
[Nov  9 14:31:29] [INFO] [  6m0s] C: 4294000/s (1511787724) N:   55000/s (19613370) W:       0/s (0) R:      0/s (0)
[Nov  9 14:32:29] [INFO] [  7m0s] C: 4371000/s (1807832315) N:   55900/s (23374305) W:       0/s (0) R:      0/s (0)
[Nov  9 14:33:29] [INFO] [  8m0s] C: 4427000/s (2092146241) N:   61700/s (29465180) W:       0/s (0) R:      0/s (0)
[Nov  9 14:34:29] [INFO] [  9m0s] C: 4472000/s (2386431349) N:   76500/s (41008548) W:       0/s (0) R:      0/s (0)
[Nov  9 14:35:29] [INFO] [ 10m0s] C: 4518000/s (2682088805) N:   75700/s (45256273) W:       0/s (0) R:      0/s (0)
[Nov  9 14:36:29] [INFO] [ 11m0s] C: 4562000/s (2979361812) N:   73900/s (48604635) W:       0/s (0) R:      0/s (0)
[Nov  9 14:37:29] [INFO] [ 12m0s] C: 4617000/s (3292239712) N:   72000/s (51642513) W:       0/s (0) R:      0/s (0)
[Nov  9 14:38:29] [INFO] [ 13m0s] C: 4655000/s (3594098633) N:   71300/s (55357611) W:       0/s (0) R:      0/s (0)
[Nov  9 14:39:29] [INFO] [ 14m0s] C: 4687000/s (3906930848) N:   70700/s (59288182) W:       0/s (0) R:      0/s (0)
[Nov  9 14:40:29] [INFO] [ 15m0s] C: 4722000/s (4212966101) N:   69000/s (61864621) W:       0/s (0) R:      0/s (0)
[Nov  9 14:41:29] [INFO] [ 16m0s] C: 4747000/s (4522371337) N:   68900/s (65929181) W:       0/s (0) R:      0/s (0)
[Nov  9 14:42:29] [INFO] [ 17m0s] C: 4775000/s (4837522516) N:   68400/s (69249526) W:       0/s (0) R:      0/s (0)
[Nov  9 14:43:29] [INFO] [ 18m0s] C: 4775000/s (4837522516) N:   68400/s (69249526) W:  941700/s (55542866) R:      0/s (0)
[Nov  9 14:44:29] [INFO] [ 19m0s] C: 4775000/s (4837522516) N:   68400/s (69249526) W:  984400/s (117141006) R:      0/s (0)
panic: runtime error: index out of range      0/s (4837522516) N:   68400/s       0/s (69249526) W:  994500/s       0/s (132759247) R:      0/s      0/s (0)

goroutine 56 [running]:
github.com/omniscale/imposm3/parser/pbf.parseTags(0xc53594e000, 0x13c03, 0x13c03, 0xc6ff6b3dc0, 0x6, 0x8, 0xc6ff633e00, 0x6, 0x8, 0xc79af7f860)
        /go/src/github.com/omniscale/imposm3/parser/pbf/pbf.go:89 +0x180
github.com/omniscale/imposm3/parser/pbf.readWays(0xc4fc2e8000, 0xb49c8, 0xde000, 0xc72cc81e00, 0xc53594e000, 0x13c03, 0x13c03, 0xc80d4c4000, 0xa9983, 0xa9983)
        /go/src/github.com/omniscale/imposm3/parser/pbf/pbf.go:151 +0x109
github.com/omniscale/imposm3/parser/pbf.(*Parser).parseBlock(0xc4200bc370, 0x7fffd223cd5b, 0x16, 0x6b8da37d2, 0x10c55e3)
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:115 +0x300
github.com/omniscale/imposm3/parser/pbf.(*Parser).Parse.func1(0xc42030a720, 0xc4200bc370)
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:56 +0xc8
created by github.com/omniscale/imposm3/parser/pbf.(*Parser).Parse
        /go/src/github.com/omniscale/imposm3/parser/pbf/process.go:65 +0x101
```

**Comment by EX0l0N** [comment]
Updating to `omniscale/imposm3` `v0.6.0-alpha.3` and its library dependencies did work, but it dies with a similar stacktrace.
I'll file a bug there.

**Comment by qlerebours** [comment]
@EX0l0N did it work on an EC2 instance for the planet ?
Can you also specify which kind of instance you used and the duration of the generation please ?

**Comment by EX0l0N** [comment]
_One does not simply_ run openmaptiles for the whole planet.

I just needed to prepare the PostGIS Database for further processing.

Running the import at AWS requires a lot of RAM for imposm3 related data structures as it easily uses up more then 16GB.
32GB should be ok.
Any number of cores will do. I used 16.

The import takes a few days resulting in a > 450GB PostGIS.

Just don't try to render the whole World from a single big PostGIS (at least not higher than zoom level 10).
The databases indexes and views become to large to be efficient.
If you run it like this it will take several months.
